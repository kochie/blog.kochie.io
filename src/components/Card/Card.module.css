.card {
  box-shadow: 8px 14px 38px rgba(39, 44, 49, 0.06), 1px 3px 8px rgba(39, 44, 49, 0.03);
  position: relative;
  /* width: max-content; */
  border-radius: 10px;
  transition: all 0.2s ease-in-out;
  background-color: var(--card-background-color);

  width: 100%;
  height: 100%;

  color: var(--heading-color);
}

.card:hover {
  opacity: 1;
}
