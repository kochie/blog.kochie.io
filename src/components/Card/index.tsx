import React, { PropsWithChildren, ReactElement } from 'react'

const Card = ({
  children,
}: PropsWithChildren<Record<never, never>>): ReactElement => {
  return (
    <div className="shadow-2xl rounded-2xl transition ease-in-out duration-200 bg-gray-300 dark:bg-neutral-600 text-black dark:text-white w-full h-full z-20 dark:shadow-neutral-800">
      {children}
    </div>
  )
}

export default Card
