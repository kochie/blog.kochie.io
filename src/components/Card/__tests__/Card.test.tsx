import React from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'

import Card from '@/components/Card'

describe('CARD COMPONENT', () => {
  test('renders correctly', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(<Card>Test</Card>)
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
