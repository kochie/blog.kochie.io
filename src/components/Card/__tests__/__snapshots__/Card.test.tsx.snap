// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CARD COMPONENT > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="shadow-2xl rounded-2xl transition ease-in-out duration-200 bg-gray-300 dark:bg-neutral-600 text-black dark:text-white w-full h-full z-20 dark:shadow-neutral-800"
  >
    Test
  </div>
</DocumentFragment>
`;
